modules/apps/client-extension/client-extension-web/src/main/resources/META-INF/resources/js/components/ScriptElementAttributes.tsx

```diff
@@ -0,0 +1,279 @@
+					onChange={handleAttributeChange}
+					type="text"
+				/>
+
+				{attrErrorMessage && (
+					<ClayForm.FeedbackGroup>
+						<ClayForm.FeedbackItem>
+							<ClayForm.FeedbackIndicator symbol="exclamation-full" />
+
+							{attrErrorMessage}
+						</ClayForm.FeedbackItem>
+					</ClayForm.FeedbackGroup>
+				)}
+			</ClayLayout.Col>
+
+			<ClayLayout.Col size={4}>
+				<label htmlFor="type">{Liferay.Language.get('type')}</label>
+
+				<Picker
+					aria-labelledby="picker-label"
+					id="type"
+					items={dataTypes}
+					onSelectionChange={(type) => {
+						if (type === 'Boolean') {
+							setAttributeValue(attributeValue);
+						}
+
+						setAttributeType(type);
+					}}
+					selectedKey={attributeType}
+				>
+					{(item) => <Option key={item}>{item}</Option>}
+				</Picker>
+			</ClayLayout.Col>
+
+			<ClayLayout.Col size={4}>
+				<div className="d-flex justify-content-between">
+					<label htmlFor="value">
+						{Liferay.Language.get('value')}
+					</label>
+
+					<div>
+						{index > 0 && (
+							<ClayButtonWithIcon
+								aria-label={Liferay.Language.get('remove')}
+								className="btn btn-primary btn-xs dm-field-repeatable-delete-button rounded-pill"
+								onClick={() => handleRemoveClick(index)}
+								symbol="hr"
+								title={Liferay.Language.get('remove')}
+								type="button"
+							/>
+						)}
+
+						<ClayButtonWithIcon
+							aria-label={Liferay.Language.get('add')}
+							className="btn btn-primary btn-xs dm-field-repeatable-add-button ml-1 rounded-pill"
+							onClick={() =>
+								attributeName
+									? handleAddClick(index)
+									: setAttrErrorMessage(
+											Liferay.Language.get(
+												'attribute-field-is-required'
+											)
+									  )
+							}
+							symbol="plus"
+							title={Liferay.Language.get('add')}
+							type="button"
+						/>
+					</div>
+				</div>
+
+				{attributeType !== 'Boolean' ? (
+					<ClayInput
+						id="value"
+						name={`${portletNamespace}value${index}`}
+						onChange={({target}) => {
+							const newValue = target.value;
+							setAttributeValue(newValue);
+						}}
+						type="text"
+					/>
+				) : (
+					<Picker
+						aria-labelledby="picker-label"
+						id="boolean"
+						items={booleanValue}
+						onSelectionChange={(value) => setAttributeValue(value)}
+						selectedKey={attributeValue.toString()}
+					>
+						{(item) => (
+							<Option key={item.value}>{item.label}</Option>
+						)}
+					</Picker>
+				)}
+			</ClayLayout.Col>
+		</ClayLayout.Row>
+	);
+};
+
+const ScriptElementAttributes = ({
+	attributesListUpdated: initialAttributesList,
+	portletNamespace,
+}) => {
+	const emptyRow = () => ({attribute: '', id: uuidv4(), type: '', value: ''});
+
+	const [attributesList, settAtributesList] = useState(
+		initialAttributesList && !!initialAttributesList.length
+			? [initialAttributesList.map((item) => ({...item, id: uuidv4()}))]
+			: [emptyRow()]
```

Comment:
You can lazy initialize the state to make it only called once in the component's life cycle. For example, even if the parameter is only used for the first rendering in subsequent renderings, the declaration will still be executed every time there is a new rendering, in this case a new list will always be generated, either empty or with initials, even if they are not used, this It's due to how the rendering system works and the nature of the functions, so to avoid this unnecessary process with each render you can involve this in slow initialization.

```jsx
const [attributesList, settAtributesList] = useState(
  () => initialAttributesList && !!initialAttributesList.length
    ? [initialAttributesList.map((item) => ({...item, id: uuidv4()}))]
    : [emptyRow()]
)
```

Reply:
Thanks Matu! 
Fixed